supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -512,7 +536,16 @@ class TesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyn
     }
 
     taskEndTime.onComplete {
-      case Success(result) => result.map(r => tellMetadata(Map(CallMetadataKeys.TaskEndTime -> r)))
+      case Success(result) =>
+        result.foreach(r => tellMetadata(Map(CallMetadataKeys.TaskEndTime -> r)))
+        val newCostData = runStatus.costData.map(_.copy(endTime = result))
+        runStatus match {
+          case _: Complete => tellBard(Complete(newCostData))
+          case _: Cancelled => tellBard(Cancelled(newCostData))
+          case failed: Failed => tellBard(Failed(sysLogs = failed.sysLogs, costData = newCostData))
+          case error: Error => tellBard(Error(sysLogs = error.sysLogs, costData = newCostData))
```

Comment:
unfortunately you cant .copy a sealed trait so this seemed like the next best thing to me.